src/rpc/net.cpp

```diff
@@ -862,23 +863,20 @@ static RPCHelpMan getnodeaddresses()
         throw JSONRPCError(RPC_CLIENT_P2P_DISABLED, "Error: Peer-to-peer functionality missing or disabled");
     }
 
-    int count = 1;
-    if (!request.params[0].isNull()) {
-        count = request.params[0].get_int();
-        if (count < 0) {
-            throw JSONRPCError(RPC_INVALID_PARAMETER, "Address count out of range");
-        }
-    }
+    const int count{request.params[0].isNull() ? 1 : request.params[0].get_int()};
+    if (count < 0) throw JSONRPCError(RPC_INVALID_PARAMETER, "Address count out of range");
+
     // returns a shuffled list of CAddress
-    std::vector<CAddress> vAddr = node.connman->GetAddresses(count, /* max_pct */ 0);
+    const std::vector<CAddress> vAddr{node.connman->GetAddresses(count, /* max_pct */ 0)};
     UniValue ret(UniValue::VARR);
 
     for (const CAddress& addr : vAddr) {
         UniValue obj(UniValue::VOBJ);
-        obj.pushKV("time", (int)addr.nTime);
-        obj.pushKV("services", (uint64_t)addr.nServices);
+        obj.pushKV("time", static_cast<int>(addr.nTime));
+        obj.pushKV("services", static_cast<uint64_t>(addr.nServices));
         obj.pushKV("address", addr.ToStringIP());
         obj.pushKV("port", addr.GetPort());
+        obj.pushKV("network", GetNetworkName(addr.GetNetClass()));
```

Comment:
Just out of curiosity: Is `ipv6` returned for both `NET_CJDNS` and `NET_IPV6`?

Reply:
It shouldn't. This essentially returns the stringified `CNetAddr::m_net`. See `netaddress.cpp::CNetAddr::SetNetFromBIP155Network()`.

Reply:
Ah, so that's the reason it is not mentioned in the help in your OP. Thanks.

Reply:
That's a different case. `GetNetworkNames()` doesn't return `cjdns` because of this line:
```cpp
if (network == NET_UNROUTABLE || network == NET_CJDNS || network == NET_INTERNAL) continue;
```

Reply:
Yes, `GetNetworkNames()` doesn't return `cjdns` because of this line:
```cpp
if (network == NET_UNROUTABLE || network == NET_CJDNS || network == NET_INTERNAL) continue;
```
When cjdns support is added, that line will need to be updated.